standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/CompactionTxnHandler.java

```diff
@@ -708,6 +734,13 @@ private void removeTxnComponents(Connection dbConn, CompactionInfo info) throws
     PreparedStatement pStmt = null;
     ResultSet rs = null;
     try {
+      /*
+       * Remove all retry associated metadata of table/partition in the TXN_CLEANUP_QUEUE both when compaction
+       * or abort cleanup is successful. We dont want a situation wherein we have a retry queue entry for a table
+       * but no corresponding entry in TXN_COMPONENTS table.
+       */
+      removeRetryQueueEntries(dbConn, info);
```

Comment:
please use `single responsibility` principle

Reply:
This does not apply since the table is removed.